src/pages/integrations/invoice/confirm-invoice/confirm-invoice.ts

```diff
@@ -0,0 +1,425 @@
+import { Component } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import {
+  Events,
+  ModalController,
+  NavController,
+  NavParams
+} from 'ionic-angular';
+import * as _ from 'lodash';
+import { Logger } from '../../../../providers/logger/logger';
+
+// Provider
+import { DecimalPipe } from '@angular/common';
+import {
+  EmailNotificationsProvider,
+  FeeProvider,
+  GiftCardProvider,
+  TxConfirmNotificationProvider,
+  WalletTabsProvider
+} from '../../../../providers';
+import { ActionSheetProvider } from '../../../../providers/action-sheet/action-sheet';
+import { AppProvider } from '../../../../providers/app/app';
+import { BwcErrorProvider } from '../../../../providers/bwc-error/bwc-error';
+import { BwcProvider } from '../../../../providers/bwc/bwc';
+import { ClipboardProvider } from '../../../../providers/clipboard/clipboard';
+import { ConfigProvider } from '../../../../providers/config/config';
+import { ExternalLinkProvider } from '../../../../providers/external-link/external-link';
+import { InvoiceProvider } from '../../../../providers/invoice/invoice';
+import { OnGoingProcessProvider } from '../../../../providers/on-going-process/on-going-process';
+import { PayproProvider } from '../../../../providers/paypro/paypro';
+import { PlatformProvider } from '../../../../providers/platform/platform';
+import { PopupProvider } from '../../../../providers/popup/popup';
+import { ProfileProvider } from '../../../../providers/profile/profile';
+import { ReplaceParametersProvider } from '../../../../providers/replace-parameters/replace-parameters';
+import { TxFormatProvider } from '../../../../providers/tx-format/tx-format';
+import {
+  Coin,
+  TransactionProposal,
+  WalletProvider
+} from '../../../../providers/wallet/wallet';
+
+// Pages
+import { ConfirmCardPurchasePage } from '../../gift-cards/confirm-card-purchase/confirm-card-purchase';
+
+@Component({
+  selector: 'confirm-invoice-page',
+  templateUrl: 'confirm-invoice.html'
+})
+export class ConfirmInvoicePage extends ConfirmCardPurchasePage {
```

Comment:
I 'd like to have `ConfirmInvoicePage` as isolated as possible, because `ConfirmCardPurchasePage` isn't the only resource that needs a BitPay invoice (see BitPayCardPage). So, I think that for now is a good idea to isolate that function that only is supported in 1/5 platforms. That way we are going to prevent changes in ConfirmCardPurchase that affect this page.